MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/kits/KitRegistry.java

```diff
@@ -56,23 +68,162 @@ public void registerKit(Kit kit) {
       plugin.getDebugger().debug("Kit " + kit.getKey() + " is disabled by kits.yml");
       return;
     }
+
     plugin.getDebugger().debug("Registered {0} kit", kit.getKey());
     kits.add(kit);
   }
 
+  /**
+   * Registers the kits by loading their configurations.
+   */
+  public void registerKits() {
+
+    for (String key : kitsConfig.getKeys(false)) {
+      if (!Objects.equals(key, "Do-Not-Edit")) {
+        loadKitConfig(key);
+      }
+    }
+  }
+
+  /**
+   * Loads the configuration for a kit.
+   *
+   * @param  kit_key  the key of the kit to load the configuration for
+   */
+  public void loadKitConfig(String kit_key) {
+    ConfigurationSection configurationSection = kitsConfig.getConfigurationSection(kit_key);
+
+    if (configurationSection == null) {
+      plugin.getDebugger().debug(Level.SEVERE, "Kit " + kit_key + " does not have any configuration.");
+      plugin.getDebugger().debug(Level.SEVERE, "Kit" + kit_key + " will not be loaded.");
+      return;
+    }
+
+    String kit_name = configurationSection.getString("name", kit_key);
+
+    if (configurationSection.getConfigurationSection("display_item") == null) {
+      configurationSection.set("display_item", XItemStack.serialize(new ItemStack(Material.BEDROCK)));
+    }
+
+    ItemStack itemStack = XItemStack.deserialize(Objects.requireNonNull(configurationSection.getConfigurationSection("display_item")));
+
+    if(!configurationSection.getBoolean("enabled", false)) {
+      plugin.getDebugger().debug("Kit " + kit_key + " is disabled by kits.yml");
+      return;
+    }
+
+    String kitType = configurationSection.getString("kit_type");
+
+    if (kitType == null) {
+      plugin.getDebugger().debug(Level.SEVERE, "Kit " + kit_key + " kit_type is null.");
+      plugin.getDebugger().debug(Level.SEVERE, "Kit" + kit_key + " will not be loaded.");
+      return;
+    }
+
+    Kit kit;
+
+    switch (kitType) {
+      case "free": {
+        kit = new FreeKit(kit_key, kit_name, itemStack);
+        break;
+      }
+      case "level": {
+        kit = new LevelKit(kit_key, kit_name, itemStack);
+        ((LevelKit) kit).setLevel(configurationSection.getInt("required-level"));
+        break;
+      }
+      case "premium": {
+        kit = new PremiumKit(kit_key, kit_name, itemStack);
+        break;
+      }
+      default: {
+        plugin.getDebugger().debug(Level.SEVERE, "Kit " + kit_key + " kit_type is not recognised.");
+        plugin.getDebugger().debug(Level.SEVERE, "Kit" + kit_key + " will not be loaded.");
+        return;
+      }
+    }
+
+    if (configurationSection.getString("unlockedOnDefault") == null) {
+      plugin.getDebugger().debug(Level.SEVERE, "Kit " + kit_key + " does not have an unlockedOnDefault configuration.");
+      plugin.getDebugger().debug(Level.SEVERE, "Kit" + kit_key + " will not be loaded.");
+      return;
+    }
+    kit.setUnlockedOnDefault(configurationSection.getBoolean("unlockedOnDefault"));
```

Comment:
I would not print a warning because of this missing `unlockedOnDefault` property, I would just specify a default value for this boolean. If this property missing from the config, it will return the default value, pretty useless and dumb users might also ask "what is this error" as usual

Reply:
unlockOnDefault is a value set for all default kits.yml

Reply:
So if you set up the kits properly, it should print anything in console.